docs/rules/group-exports.md

```diff
@@ -0,0 +1,87 @@
+# group-exports
+
+Reports when named exports are not grouped together in a single `export` declaration or when multiple assignments to CommonJS `module.exports` or `exports` object are present in a single file.
+
+**Rationale:** An `export` declaration or `module.exports` assignment can appear anywhere in the code. By requiring a single export declaration all your exports will remain at one place, making it easier to see what exports a module provides.
+
+## Rule Details
+
+This rule warns whenever a single file contains multiple named export declarations or multiple assignments to `module.exports` (or `exports`).
```

Comment:
I do not currently see any issues with auto-fixing, either, but I have not included a fixer in this PR for several reasons:

- I am kind of new to ESLint rule development
- I have no prior experience manipulating or generally working with AST
- I wanted to keep this small and push it out fast (even though I then got lost in other tasks and forgot about this PR for several months 😄 )

So I'd rather not add fixer at this moment, even though it should be possible to implement it, eventually.